src/SME.SGP.Aplicacao/Comandos/ComandosPlanoAula.cs

```diff
@@ -196,28 +202,37 @@ private PlanoAula MapearParaDominio(PlanoAulaDto planoDto, PlanoAula planoAula =
             return planoAula;
         }
 
-        private void ValidarMigracao(MigrarPlanoAulaDto migrarPlanoAulaDto)
+        private async Task ValidarMigracao(MigrarPlanoAulaDto migrarPlanoAulaDto, string codigoRf, bool ehProfessorCj, string ueId)
         {
-            var turmasAtribuidasAoProfessor = consultasProfessor.Listar(migrarPlanoAulaDto.RFProfessor);
+            var turmasAtribuidasAoProfessor = consultasProfessor.Listar(codigoRf);
             var idsTurmasSelecionadas = migrarPlanoAulaDto.IdsPlanoTurmasDestino.Select(x => x.TurmaId).ToList();
 
-            ValidaTurmasProfessor(migrarPlanoAulaDto.EhProfessorCJ,
+            await ValidaTurmasProfessor(ehProfessorCj, ueId, 
+                                  migrarPlanoAulaDto.DisciplinaId, 
+                                  codigoRf,
                                   turmasAtribuidasAoProfessor,
                                   idsTurmasSelecionadas);
 
-            ValidaTurmasAno(migrarPlanoAulaDto.EhProfessorCJ, migrarPlanoAulaDto.MigrarObjetivos,
+            ValidaTurmasAno(ehProfessorCj, migrarPlanoAulaDto.MigrarObjetivos,
                             turmasAtribuidasAoProfessor, idsTurmasSelecionadas);
         }
 
-        private void ValidaTurmasProfessor(bool ehProfessorCJ,
+        private async Task ValidaTurmasProfessor(bool ehProfessorCJ, string ueId, string disciplinaId, string codigoRf,
                                            IEnumerable<ProfessorTurmaDto> turmasAtribuidasAoProfessor,
                                            IEnumerable<string> idsTurmasSelecionadas)
         {
             var idsTurmasProfessor = turmasAtribuidasAoProfessor?.Select(c => c.CodTurma).ToList();
 
             if (ehProfessorCJ)
             {
-                //validar se o professor CJ possui atribuição de disciplina no ano
+                foreach (var idTurma in idsTurmasSelecionadas)
+                {
+                    IEnumerable<AtribuicaoCJ> lstTurmasCJ = await
+                         repositorioAtribuicaoCJ.ObterPorFiltros(null, idTurma, ueId, Convert.ToInt64(disciplinaId), codigoRf, null, null);
```

Comment:
isso performaticamente não vai ficar legal. 
Não seria melhor obter a lista das disciplinas que aquele professor está substuindo(adicionei outro filtro dizendo se é para buscar os que estão substituindo) fora do 'for' e dentro tu só verificar se aquele disciplina está na lista?